drivers/modem/gsm_ppp.c

```diff
@@ -353,6 +400,11 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
 {
 	int ret;
 
+	/* If modem is searching for network, we should skip the setup step */
+	if ((gsm->net_state == GSM_SEARCHING) && gsm->register_retries) {
+		goto registering;
+	}
```

Comment:
there's no sense to check registration before attachment as attachment comes first so as for me attachment check can be replaced by the registration check and that's it.

Reply:
I'm not sure about this one, is it possible where a modem registered to a tower but couldn't attach to a packet service?